File: scanner_exe/backend/app/types/scanner.py
from typing import Optional, List
from pydantic import BaseModel


class ScanRequest(BaseModel):
    scanner_id: str
    resolution: Optional[int] = 300
    color_mode: Optional[str] = "color"  # color, grayscale, or black_and_white
    client_id: Optional[str] = None

class ScanResponse(BaseModel):
    success: bool
    message: str
    image_data: Optional[str] = None  # Base64 encoded image

class Scanner(BaseModel):
    id: str
    name: str
    manufacturer: str
    model: str
    type: Optional[str] = None

class ListScannersResponse(BaseModel):
    action: str = "list_scanners"
    scanners: List[Scanner]

File: scanner_exe/frontend/src/types/scanner.ts
export interface Scanner {
  id: string;
  name: string;
  manufacturer?: string;
  model?: string;
}

export interface ScanRequest {
  action: string;
  data: {
    scanner_id: string;
    resolution: number;
    color_mode: string;
    client_id?: string;
  };
}

export interface ScanResponse {
  action?: string;
  status: string;
  message?: string;
  image_data?: string;
  format?: string;
  demo?: boolean;
  success?: boolean;
}
